### PR TITLE
Add email config for transferEncoding #11485

### PR DESCRIPTION
**Pull request by maratth**
Add getter/setter for transferEncoding and modify the `_getContentTranferEncoding` for use transferEncoding if is set.

Write test in `EmailTest` for check it.

### CODE REVIEW

File: tests/TestCase/Mailer/EmailTest.php [hunk 1]
```diff
@@ -2487,6 +2497,23 @@ public function testHeaderCharset()
...
 
+    /**
+     * Test transferEncoding
+     *
+     * @return void
+     */
+    public function testTransferEncoding(){
+        // Test new transfer encoding
+        $this->Email->setTransferEncoding('quoted-printable');
+        $this->assertSame($this->Email->getTransferEncoding(), 'quoted-printable');
```
Comment (dereuromark) [inline tests/TestCase/Mailer/EmailTest.php]: Order is wrong, should be: expected, result.

File: src/Mailer/Email.php [hunk 2]
```diff
@@ -241,6 +241,14 @@ class Email implements JsonSerializable, Serializable
      */
     public $headerCharset;
 
+    /**
+     * The email transfer encoding used.
+     * If null, the $charset property is used for determined the transfer encoding.
+     *
+     * @var string|null
+     */
+    public $transferEncoding;
```
Comment (robertpustulka) [inline src/Mailer/Email.php]: Does it make sense to add another public properties and a pair of getter/setter for that?

Comment (LifeOrYou) [reply src/Mailer/Email.php]: I think no, but i follow what was done. 
You can see the charset or headerCharset properties.

Comment (robertpustulka) [reply src/Mailer/Email.php]: Those are public for BC reasons I suppose. I think we can make the new property protected.

File: tests/TestCase/Mailer/EmailTest.php (hunk 1 above)
Comment (LifeOrYou) [reply tests/TestCase/Mailer/EmailTest.php]: Hum, sorry i fix this in #a006420
But i see, the tests above have the same issue.

File: src/Mailer/Email.php (hunk 2 above)
Comment (ADmad) [reply src/Mailer/Email.php]: Yeah the property should be protected.

Comment (LifeOrYou) [reply src/Mailer/Email.php]: Ok i fix it

File: src/Mailer/Email.php [hunk 3]
```diff
@@ -827,6 +835,29 @@ public function headerCharset($charset = null)
...
 
+    /**
+     * TransferEncoding setter.
+     *
+     * @param string|null $encoding Encoding set.
+     * @return $this
+     */
+    public function setTransferEncoding($encoding)
+    {
+        $this->transferEncoding = $encoding;
```
Comment (markstory) [inline src/Mailer/Email.php]: Should we validate that for a valid/known transfer encoding?

Comment (LifeOrYou) [reply src/Mailer/Email.php]: I don't know all content transfer encoding. But I found [this](https://www.w3.org/Protocols/rfc1341/5_Content-Transfer-Encoding.html). 
I understand that there is `BASE64 / QUOTED-PRINTABLE / 8BIT / 7BIT / BINARY / x-token` but I'm not sure, I'm not a expert.

If you think is good, I can implement it.

Comment (markstory) [reply src/Mailer/Email.php]: I think checking the standard types (not `x-token`) is a good place to start.

**Comment by LifeOrYou** [comment]
Is possible add this pull request for cakephp v3.4.x or is only for 3.5.x ?

**Comment by markstory** [comment]
> Is possible add this pull request for cakephp v3.4.x or is only for 3.5.x ?

We're only doing security fixes for 3.4 at this point in time. The upgrade from 3.4 to 3.5 should be pretty painless if you need this in your app.

File: src/Mailer/Email.php [hunk 4]
```diff
@@ -827,6 +848,38 @@ public function headerCharset($charset = null)
         return $this->headerCharset;
     }
 
+    /**
+     * TransferEncoding setter.
+     *
+     * @param string|null $encoding Encoding set.
+     * @return $this
+     */
+    public function setTransferEncoding($encoding)
```
Comment (markstory) [inline src/Mailer/Email.php]: Should we be handling encoding the message based on the indicated transfer encoding? Right now someone could set the transfer encoding to base64, but the email message will still be in 8bit (generally).

Comment (LifeOrYou) [reply src/Mailer/Email.php]: I think than that two differents parts. I don't know if have a link between transfer encoding and message encoding.

Comment (markstory) [reply src/Mailer/Email.php]: Ok.

**Pull request merged**